jb/detail/array_traits.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef jb_detail_array_traits_hpp
+#define jb_detail_array_traits_hpp
+
+#include <boost/multi_array.hpp>
+#include <vector>
+
+namespace jb {
+namespace detail {
+/**
+ * Define traits for array-like containers.
+ *
+ * The generic version implements the class assuming std::vector<> or
+ * any similar container.  We provide specializations for other types
+ * (such as boost::multi_array<>) to be able to operate on them
+ * generically.
+ *
+ * @tparam container_type the type of container, typically std::vector<>
+ */
+template <typename container_type>
+struct array_traits {
+  /// Define the type of the elements in the container
+  using element_type = typename container_type::value_type;
+};
+
+/// Count the number of elements for a vector-like container
```

Comment:
This function needs a longer documentation, model it after the previous class (for example).